### PR TITLE
XML doc comments for System.Linq.Expressions

### PR DESCRIPTION
**Pull request by bartdesmet**
These were in a bit of an intermediate state, with typos here and there, some missing ones, and inconsistencies in the way `cref` attributes were specified. I did a bunch of search-and-replace operations to fix them up and filled in the gaps by taking documentation from MSDN. Also changing the `cref` style to omit `T:`, `P:`, `M:`, `F:` prefixes, which enables the C# compiler to check the references properly; this uncovered a bunch of incorrect references to non-existing types or members.

I'm not sure what the plan is with regards to XML documentation and building it to produce .NET Core documentation online. For this assembly, with these changes, we can almost enable the XML doc generation in the `.csproj` file. There are a few more errors on ambiguous references in `cref` attributes and missing XML docs on some public members that should not be public (I'll submit a separate PR for that).

CC @stephentoub

### CODE REVIEW

File: src/System.Linq.Expressions/src/System/Linq/IQueryable.cs [hunk 1]
```diff
@@ -8,34 +8,99 @@
 
 namespace System.Linq
 {
+    /// <summary>
+    /// Provides functionality to evaluate queries against a specific data source wherein the type of the data is not specified.
+    /// </summary>
     public interface IQueryable : IEnumerable
     {
+        /// <summary>
+        /// Gets the expression tree that is associated with the instance of IQueryable.
```
Comment (stephentoub) [inline src/System.Linq.Expressions/src/System/Linq/IQueryable.cs]: Should these references to IQueryable be in see tags?

File: src/System.Linq.Expressions/src/System/Linq/Expressions/DefaultExpression.cs [hunk 2]
```diff
@@ -21,9 +21,9 @@ internal DefaultExpression(Type type)
         }
 
         /// <summary>
-        /// Gets the static type of the expression that this <see cref="Expression" /> represents.
+        /// Gets the static type of the expression that this <see cref="Expression"/> represents.
         /// </summary>
-        /// <returns>The <see cref="Type"/> that represents the static type of the expression.</returns>
+        /// <returns>The <see cref="System.Type"/> that represents the static type of the expression.</returns>
```
Comment (stephentoub) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/DefaultExpression.cs]: Good catch.  Did you find this, or is this something that was somehow caught by an analyzer?

File: src/System.Linq.Expressions/src/System/Linq/Expressions/IArgumentProvider.cs [hunk 3]
```diff
@@ -5,10 +5,10 @@
 namespace System.Linq.Expressions
 {
     /// <summary>
-    /// Provides an internal interface for accessing the arguments that multiple tree
-    /// nodes (DynamicExpression, ElementInit, MethodCallExpression, InvocationExpression, NewExpression,
-    /// and InexExpression).
-    /// 
+    /// Interface for accessing the arguments of multiple tree nodes (DynamicExpression, ElementInit,
+    /// MethodCallExpression, InvocationExpression, NewExpression, and IndexExpression).
```
Comment (stephentoub) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/IArgumentProvider.cs]: Should all of these type references be in see tags?

**Comment by stephentoub** [comment]
:shipit:

File: src/System.Linq.Expressions/src/System/Linq/Expressions/DefaultExpression.cs (hunk 2 above)
Comment (bartdesmet) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/DefaultExpression.cs]: Manual at this point.

File: src/System.Linq.Expressions/src/System/Linq/Expressions/IArgumentProvider.cs (hunk 3 above)
Comment (bartdesmet) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/IArgumentProvider.cs]: Yes, with the exception of `DynamicExpression` which lives in another assembly, but I could use the interface type there instead.

**Pull request merged**